aten/src/ATen/native/nestedtensor/NestedTensorImpl.cpp

```diff
@@ -0,0 +1,110 @@
+#include <ATen/ATen.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/WrapDimUtils.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/nestedtensor/NestedTensorImpl.h>
+#include <c10/core/DispatchKey.h>
+
+namespace at {
+namespace native {
+
+NestedTensorImpl::NestedTensorImpl(
+    at::Tensor buffer,
+    at::Tensor nested_size_tensor)
+    : TensorImpl(
+          c10::DispatchKeySet({DispatchKey::NestedTensor}),
+          buffer.dtype(),
+          buffer.device()),
+      _buffer(std::move(buffer)),
+      _nested_size_tensor(std::move(nested_size_tensor)) {
```

Comment:
any requirements on nested_size_tensor? Looks like it has to be contiguous to me..